File: backend/app/api/routes/recovery.py
from fastapi import APIRouter

from app.decision.engine import decide_recovery_action
from app.agent.agent import agent_singleton
from app.agent.schemas import AgentResponse
from app.schemas.recovery import RecoveryDecisionRequest, RecoveryDecisionResponse, RecoveryRequest

router = APIRouter(prefix="/api/v1/recovery", tags=["recovery"])


@router.post("/decision", response_model=RecoveryDecisionResponse)
def create_recovery_decision(request: RecoveryDecisionRequest) -> dict:
    order = {
        "order_id": request.order_id,
        "amount": request.amount,
        "attempt_count": request.attempt_count,
    }
    return decide_recovery_action(order=order, rto_probability=request.rto_probability)


@router.post("/request", response_model=AgentResponse)
def request_recovery(request: RecoveryRequest) -> AgentResponse:
    return agent_singleton.request_recovery(request.order_id, request.action, request.session_id)

File: frontend/src/lib/api/recovery.ts
import { apiFetch } from "./client";
import type {
  AgentResponse,
  RecoveryDecisionRequest,
  RecoveryDecisionResponse,
  RecoveryRequest,
} from "../types";

export async function getRecoveryDecision(
  request: RecoveryDecisionRequest,
): Promise<RecoveryDecisionResponse> {
  return apiFetch<RecoveryDecisionResponse>("/api/v1/recovery/decision", {
    method: "POST",
    body: JSON.stringify(request),
  });
}

export async function requestRecovery(request: RecoveryRequest): Promise<AgentResponse> {
  return apiFetch<AgentResponse>("/api/v1/recovery/request", {
    method: "POST",
    body: JSON.stringify(request),
  });
}
